tests/python/test_with_dask.py

```diff
@@ -0,0 +1,73 @@
+import testing as tm
+from distributed import Client, LocalCluster
+import dask.dataframe as dd
+import xgboost as xgb
+import numpy as np
+import pytest
+
+pytestmark = pytest.mark.skipif(**tm.no_sklearn())
+
+
+def run_train():
+    # Contains one label equal to rank
+    dmat = xgb.DMatrix([[0]], label=[xgb.rabit.get_rank()])
+    bst = xgb.train({"eta": 1.0}, dmat, 1)
+    return bst.predict(dmat)
+
+
+def test_train():
+    # Train two workers, the first has label 0, the second has label 1
+    # If they build the model together the output should be 0.5
+    cluster = LocalCluster(n_workers=2, threads_per_worker=2)
+    client = Client(cluster)

```

Comment:
You may want to take a look at our testing infrastructure: https://distributed.dask.org/en/latest/develop.html#writing-tests

These objects create and destroy processes and event loops.  If you do this a few times in CI systems then things can get a little messy.  Those testing harnesses should keep things relatively clean.  In particular you might want the `client` pytest fixture.


```python
from distributed.utils_test import client

def test_train(client):
    preds = xgb.dask.run(client, run_train)
    assert all(v[0] == 0.5 for v in preds.values())
````

Reply:
When I do this I get `AttributeError: 'function' object has no attribute 'run_on_scheduler'

Reply:
I recommend starting from here:

```python
from distributed.utils_test import client, loop, cluster_fixture
from dask.distributed import Client

def test_client_fixture(client):
    assert isinstance(client, Client)
    assert hasattr(client, 'run_on_scheduler')
```

```
py.test test_foo.py 
=============================================== test session starts ===============================================
platform linux -- Python 3.6.7, pytest-4.1.1, py-1.7.0, pluggy-0.8.0 -- /home/mrocklin/Software/anaconda/bin/python
cachedir: .pytest_cache
rootdir: /home/mrocklin/workspace/distributed, inifile: setup.cfg
plugins: xdist-1.23.2, timeout-1.2.1, forked-0.2, cov-2.5.1, asyncio-0.10.0
collected 1 item                                                                                                  

baz.py::test_client_fixture PASSED                                                                          [100%]

============================================ slowest 10 test durations ============================================
0.80s setup    baz.py::test_client_fixture
0.52s teardown baz.py::test_client_fixture

(0.00 durations hidden.  Use -vv to show these durations.)
============================================ 1 passed in 1.41 seconds =============================================
```

I should have mentioned that this assumes that you're using pytest, which I recognize now may not be a valid assumption.